src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.InlineContext.cs

```diff
@@ -0,0 +1,582 @@
+                IArgumentOperation argumentOperation)
+            {
+                var parameterSymbol = argumentOperation.Parameter;
+                var name = renameTable.ContainsKey(parameterSymbol) ? renameTable[parameterSymbol] : parameterSymbol.Name;
+                // Check has been done before to make sure there is only one child.
+                var argumentExpressionOperation = argumentOperation.Children.First();
+                if (argumentOperation.ArgumentKind == ArgumentKind.ParamArray
+                    && argumentExpressionOperation is IArrayCreationOperation arrayCreationOperation
+                    && argumentOperation.IsImplicit)
+                {
+                    // if this argument is a param array & the array creation operation is implicitly generated,
+                    // it means it is in this format:
+                    // void caller() { Callee(1, 2, 3); }
+                    // void Callee(params int[] x) { }
+                    // Collect each of these arguments and generate a new array for it.
+                    // Note: it could be empty.
+                    return syntaxGenerator.LocalDeclarationStatement(
+                        parameterSymbol.Type,
+                        name,
+                        inlineMethodRefactoringProvider.GenerateArrayInitializerExpression(arrayCreationOperation.Initializer.ElementValues.SelectAsArray(op => op.Syntax)));
+                }
+                else
+                {
+                    // In all the other cases, one parameter should only maps to one arguments.
+                    return syntaxGenerator.LocalDeclarationStatement(parameterSymbol.Type, name, argumentExpressionOperation.Syntax);
+                }
+            }
+
+            private static async Task<SyntaxNode> ReplaceAllSyntaxNodesForSymbolAsync(
+                Document document,
+                TExpression inlineNode,
+                SyntaxGenerator syntaxGenerator,
+                SyntaxNode root,
+                ImmutableDictionary<ISymbol, SyntaxNode> replacementTable,
+                CancellationToken cancellationToken)
+            {
+                var editor = new SyntaxEditor(inlineNode, syntaxGenerator);
+
+                foreach (var kvp in replacementTable)
+                {
+                    var allReferences = await SymbolFinder
+                        .FindReferencesAsync(kvp.Key, document.Project.Solution, cancellationToken)
+                        .ConfigureAwait(false);
+                    var allSyntaxNodesToReplace = allReferences
+                        .SelectMany(reference => reference.Locations
+                            .Select(location => root.FindNode(location.Location.SourceSpan)))
+                        .ToImmutableArray();
+
+                    foreach (var nodeToReplace in allSyntaxNodesToReplace)
+                    {
+                        if (editor.OriginalRoot.Contains(nodeToReplace))
```

Comment:
why do you need this check.

are you just finding references to locals?

Reply:
I found one case which the reference it not limited within the local body.
`Caller() { Callee(i: 100); }`
`Callee(int i) { // Do Something}`
Then the 'i' in the caller is considered also as a references, that that's why it has this check